healthcheck.go

```diff
@@ -1,118 +1,78 @@
-	return fthealth.HandlerParallel("Dependent services healthcheck", "Checks if all the dependent services are reachable and healthy.", h.messageQueueProxyReachable())
+func NewHealthCheck(p producer.MessageProducer, c consumer.MessageConsumer) *HealthCheck {
+	return &HealthCheck{
+		consumer: c,
+		producer: p,
+	}
 }
 
-func (h *Healthcheck) gtg(writer http.ResponseWriter, req *http.Request) {
-	healthChecks := []func() error{h.checkAggregateMessageQueueProxiesReachable}
-
-	for _, hCheck := range healthChecks {
-		if err := hCheck(); err != nil {
-			writer.WriteHeader(http.StatusServiceUnavailable)
-			return
-		}
+func (h *HealthCheck) Health() func(w http.ResponseWriter, r *http.Request) {
+	checks := []fthealth.Check{h.readQueueCheck(), h.writeQueueCheck()}
+	hc := fthealth.HealthCheck{
+		SystemCode:  "v1-suggestor",
+		Name:        "V1 Suggestor",
+		Description: "Checks if all the dependent services are reachable and healthy.",
+		Checks:      checks,
 	}
+	return fthealth.Handler(hc)
 }
 
-func (h *Healthcheck) messageQueueProxyReachable() fthealth.Check {
+func (h *HealthCheck) readQueueCheck() fthealth.Check {
 	return fthealth.Check{
-		BusinessImpact:   "Content V1 Metadata is not suggested. This will negatively impact V1 metadata availability.",
-		Name:             "MessageQueueProxyReachable",
-		PanicGuide:       "https://sites.google.com/a/ft.com/technology/systems/dynamic-semantic-publishing/extra-publishing/v1-suggestor-runbook",
+		ID:               "read-message-queue-proxy-reachable",
+		Name:             "Read Message Queue Proxy Reachable",
 		Severity:         1,
-		TechnicalSummary: "Message queue proxy is not reachable/healthy",
-		Checker:          h.checkAggregateMessageQueueProxiesReachable,
+		BusinessImpact:   "Content V1 Metadata is not suggested. This will negatively impact V1 metadata availability.",
+		TechnicalSummary: "Read message queue proxy is not reachable/healthy",
+		PanicGuide:       "https://dewey.ft.com/",
+		Checker:          h.consumer.ConnectivityCheck,
 	}
-
 }
 
-func (h *Healthcheck) checkAggregateMessageQueueProxiesReachable() error {
-
-	errMsg := ""
-
-	for i := 0; i < len(h.srcConf.Addrs); i++ {
-		err := h.checkMessageQueueProxyReachable(h.srcConf.Addrs[i], h.srcConf.Topic, h.srcConf.AuthorizationKey, h.srcConf.Queue)
-		if err == nil {
-			return nil
-		}
-		errMsg = errMsg + fmt.Sprintf("For %s there is an error %v \n", h.srcConf.Addrs[i], err.Error())
-	}
-
-	err := h.checkMessageQueueProxyReachable(h.destConf.Addr, h.destConf.Topic, h.destConf.Authorization, h.destConf.Queue)
-	if err == nil {
-		return nil
+func (h *HealthCheck) writeQueueCheck() fthealth.Check {
+	return fthealth.Check{
+		ID:               "write-message-queue-proxy-reachable",
+		Name:             "Write Message Queue Proxy Reachable",
+		Severity:         1,
+		BusinessImpact:   "Content V1 Metadata is not suggested. This will negatively impact V1 metadata availability.",
```

Comment:
Business Impact message is not clear enough. Suggestion: "Content V1 Metadata can't be propagated through the stack. V1 metadata won't be available for new content."